src/test/java/design/subsite/CellTest.java

```diff
@@ -0,0 +1,320 @@
+         verifyInputPins(iportcell, Arrays.asList());
+         verifyInputPins(oportcell, Arrays.asList("PAD"));
+         verifyInputPins(ioportcell, Arrays.asList("PAD"));
+         verifyInputPins(gndcell, Arrays.asList());
+         verifyInputPins(vcccell, Arrays.asList());
+     }
+
+     /**
+      * helper function to test a Cell to make sure it has the given output pins
+      * cell the Cell to test
+      * expected the list of output pin names the cell should have
+      */
+     private void verifyOutputPins(Cell cell, List<String> expected) {
+         List<String> actual = cell.getOutputPins().stream()
+                 .map(CellPin::getName)
+                 .collect(Collectors.toList());
+         assertEquals(expected.size(), actual.size(), "Expected output pin count for " + cell.getName() + " Cell doesn't match calculated.");
+         for (String pin : expected) {
+             assertTrue(actual.contains(pin), cell.getName() + " Cell doesn't have " + pin + " output pin.");
+         }
+     }
+
+     /**
+      * helper function to test a Cell to make sure it has the given input pins
+      * cell the Cell to test
+      * pins the list of input pin names the cell should have
+      */
+     private void verifyInputPins(Cell cell, List<String> expected) {
+         List<String> actual = cell.getInputPins().stream()
+                 .map(CellPin::getName)
+                 .collect(Collectors.toList());
+         assertEquals(expected.size(), actual.size(), "Expected input pin count for " + cell.getName() + " Cell doesn't match calcualted.");
+         for (String pin : expected) {
+             assertTrue(actual.contains(pin), cell.getName() + " Cell doesn't have " + pin + " input pin.");
+         }
+     }
+
+     @Test
+     @DisplayName("test 'Dir' property of PORT Cells.")
+     public void testDirProperty() {
+         verifyDirProperty(iportcell, PortDirection.IN);
+         verifyDirProperty(oportcell, PortDirection.OUT);
+         verifyDirProperty(ioportcell, PortDirection.INOUT);
+     }
+
+     /**
+      * helper function to test a Cell for a certain value of the Dir Property
+      * cell the Cell to test
+      * property the Property to check for
+      */
+     private void verifyDirProperty(Cell cell, PortDirection property) {
+         assertTrue(cell.getProperties().has("Dir"), cell.getName() + " Cell doesn't have 'Dir' property.");
+         assertEquals(property, cell.getProperties().get("Dir").getValue(), cell.getName() + " Cell has improper 'Dir' property value.");
+     }
+
+     @Test
+     @DisplayName("test Cell method 'getPossibleAnchors'")
+     public void testGetPossibleAnchors() {
+         verifyAnchors(lutcell, Arrays.asList("SLICEL-D6", "SLICEL-D5", "SLICEL-C6", "SLICEL-C5", "SLICEL-B6", "SLICEL-B5", "SLICEL-A5", "SLICEL-A6", "SLICEM-D6", "SLICEM-D5", "SLICEM-C6", "SLICEM-C5", "SLICEM-B6", "SLICEM-B5", "SLICEM-A5", "SLICEM-A6"));
```

Comment:
I don't expect the runtime to be much of an issue. The reason that I used a List was simply because of the easy constructor: Arrays.asList